### PR TITLE
Introduce ability to pass custom storage as a prop. defaults to local storage.

### PR DESCRIPTION
**Pull request by notgiorgi**
Introduce ability to pass custom storage as a prop. defaults to local storage.

I'll add docs after the review.

### CODE REVIEW

File: src/Storages/LocalStorage.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Storage from './Storage';
+
+export default class LocalStorage implements Storage {
```
Comment (jaredpalmer) [inline src/Storages/LocalStorage.ts]: Used named exports everywhere please. no defaults

File: src/Storages/Storage.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default interface Storage {
```
Comment (jaredpalmer) [inline src/Storages/Storage.ts]: use named export

File: src/Storages/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export { default as Storage } from './Storage';
```
Comment (jaredpalmer) [inline src/Storages/index.ts]: export * from './Storage'
export * from './LocalStorage'

File: src/formik-persist.tsx [hunk 4]
```diff
@@ -31,8 +38,10 @@ export class Persist extends React.Component<PersistProps, {}> {
     }
   }
 
-  componentDidMount() {
-    const maybeState = window.localStorage.getItem(this.props.name);
+  async componentDidMount() {
```
Comment (jaredpalmer) [inline src/formik-persist.tsx]: Using async here can lead to inconsistent state. Stick with`.then(success => {}, error => {})`

**Comment by notgiorgi** [comment]
@jaredpalmer Hi, I fixed the exports and async/await stuff.

Not sure how to proceed with

>So all in all I'm not sure passing storage as a prop is a good API. I think i'd rather expose a couple of the most common persistence options as components and the a factory to make your own. So<Persist > takes a prop like storage="localStorage" or storage="asyncStorage"

Should I introduce react-native as a peer dependency?

**Comment by lucasbento** [comment]
This is a good PR, I would go with tweaking it to work with implementing `AsyncStorage` in the code (as it is on `LocalStorage` file) and having the prop `storage="asyncStorage"` as per what @jaredpalmer said.

Just bumping this since it's pretty useful for me.

**Pull request closed without merging**